lisa/transformers/kernel_installer.py

```diff
@@ -138,10 +138,18 @@ def _internal_run(self) -> Dict[str, Any]:
         self._log.info(f"installed kernel version: {installed_kernel_version}")
 
         # for ubuntu cvm kernel, there is no menuentry added into grub file
-        if hasattr(installer.runbook, "source"):
-            if installer.runbook.source != "linux-image-azure-fde":
-                posix = cast(Posix, node.os)
-                posix.replace_boot_kernel(installed_kernel_version)
+        # when the installer's type is "source", it needs to add the menuentry into
+        # grub file. Otherwise, the node might not boot into the new kernel especially
+        # the new kernel is not the latest version.
```

Comment:
does it mean the installed kernel version is lower than current kernel version?

Reply:
Yes. I have modified the comment. Your description is much clearer.